File: backend/src/main/java/org/kfokam48/stagemanagementbackend/service/SecteurService.java
package org.kfokam48.stagemanagementbackend.service;

import org.kfokam48.stagemanagementbackend.dto.secteur.SecteurRequestDTO;
import org.kfokam48.stagemanagementbackend.model.Secteur;

import java.util.List;

public interface SecteurService {
    public Secteur getSecteurById(Long id);
    public List<Secteur> getAllSecteurs();
    public Secteur saveSecteur(SecteurRequestDTO secteurRequestDTO);
    public String deleteSecteur(Long id);
    public Secteur updateSecteur(Long id,SecteurRequestDTO secteurRequestDTO);
}

File: lib/secteur-service.ts
import { apiService } from './api'

export interface Secteur {
  id: number
  nomSecteur: string
}

export interface SecteurRequest {
  nomSecteur: string
}

export const secteurService = {
  async getAllSecteurs(): Promise<Secteur[]> {
    const response = await apiService.get('/api/v1/secteurs')
    return response.json()
  },

  async getSecteurById(id: number): Promise<Secteur> {
    const response = await apiService.get(`/api/v1/secteurs/${id}`)
    return response.json()
  },

  async createSecteur(data: SecteurRequest): Promise<Secteur> {
    const response = await apiService.post('/api/v1/secteurs', data)
    return response.json()
  },

  async updateSecteur(id: number, data: SecteurRequest): Promise<Secteur> {
    const response = await apiService.put(`/api/v1/secteurs/${id}`, data)
    return response.json()
  },

  async deleteSecteur(id: number): Promise<string> {
    const response = await apiService.delete(`/api/v1/secteurs/${id}`)
    return response.json()
  }
}